demos/demo_teststand.cpp

```diff
@@ -17,7 +17,9 @@ using namespace blmc_robots;
 
 static THREAD_FUNCTION_RETURN_TYPE control_loop(void* robot_void_ptr)
 {
-    Teststand& robot = *(static_cast<Teststand*>(robot_void_ptr));
```

Comment:
When you create a real time thread it is strongly discourage to create the memory inside the thread. Do you have a reasonable reason for doing this? Otherwise please revert this change and create the robot in the main.

Reply:
> When you create a real time thread it is strongly discourage to create the memory inside the thread.

Can you explain why? I understand you shouldn't allocate on the main control loop but I don't see why doing it at the beginning of the rt thread before doing control is a problem.

Reply:
> Do you have a reasonable reason for doing this?

Similar like what I found when using the master board sdk: When initializing the object on the main thread and then start using the object passed via a pointer from the rt thread, things don't work. A similar behavior happend when we tried to read from the IMU.

We should investigate this further but I was not sure if we need to do it in this PR or not.

@MaximilienNaveau, @vincentberenz , can you look into why init from main thread and using from rt thread is causing problems? In our case we init the object and then spawn the rt thread. I wonder if spawning the rt thread takes some time. If the device expects to have constant communication after initialization, then this delay might cause the communication to break. This could explain what we observe.

Reply:
I can check, but the imu code is simple, upon init I start a real time thread that reads the imu. When you do a get on the imu object you simply copy the last read results. so the communication is never interrupted. Except if creating a real time thread preempt everything else...

Reply:
> > When you create a real time thread it is strongly discourage to create the memory inside the thread.
> 
> Can you explain why? I understand you shouldn't allocate on the main control loop but I don't see why doing it at the beginning of the rt thread before doing control is a problem.

This is what is recommended by the official rt_preempt repos in their development guide. They admit that creating the memory in the real time thread should not cause too much trouble but they advice against doing it. Ideally the real time thread just do some operation into some already initialized objects.
Last paragraph:
https://wiki.linuxfoundation.org/realtime/documentation/howto/applications/memory#memory_locking

Reply:
Following recommendations is good. However, it doesn't work all the time. For instance, for tensorflow, I had to move the main initialization (and memory allocation) inside the rt_thread as well. When I tried to initialize the tensorflow object on the main thread and use the allocated objects from the rt_thread, things crashed.

See this commit: https://git-amd.tuebingen.mpg.de/amd-clmc/dg_tensorflow/commit/fc94de7fdc179d29f2a22975b198a7f458aaff9a